responsemanager/queryexecutor/queryexecutor.go

```diff
@@ -195,18 +196,28 @@ func (qe *QueryExecutor) runTraversal(p peer.ID, taskData ResponseTask) error {
 			}
 			return err
 		}
-		lnk, data, err := qe.nextBlock(taskData)
+		lnk, _ := taskData.Traverser.CurrentRequest()
+		ctx, span := otel.Tracer("graphsync").Start(ctx, "processBlock", trace.WithAttributes(
```

Comment:
why not defer the `span.End()` for this one rather than have 3 separate calls for it?

Reply:
cause it's a for-loop. golang defer is function scoped